tests/cxx/using_overload-float.h

```diff
@@ -7,7 +7,11 @@
 //
 //===----------------------------------------------------------------------===//
 
+#ifndef INCLUDE_WHAT_YOU_USE_TESTS_CXX_USING_OVERLOAD_FLOAT_H_
```

Comment:
I'd prefer not to introduce coupling between tests by reusing this header (it appears to be part of `using_overload_{multiple,single}.cc`). But I see that it's annoying to add new headers for these nonsense decls. I also can't see any idiomatic way to add reusable two-part headers (we have `direct.h`/`indirect.h` that are sanctioned to reuse between tests, but there's no precedent for numbering there, I think).

Reply:
There is no problem with adding specific headers. I was just concerned about `tests/cxx` folder bloating.

Reply:
Yeah, it's getting a bit crowded. We should probably think about further subdivision at some point, but let's keep them separated for now.